linera-views/src/memory.rs

```diff
@@ -193,84 +201,109 @@ impl WritableKeyValueStore<MemoryStoreError> for MemoryStore {
         Ok(())
     }
 
-    async fn clear_journal(&self, _base_key: &[u8]) -> Result<(), MemoryStoreError> {
+    async fn clear_journal(&self) -> Result<(), MemoryStoreError> {
         Ok(())
     }
 }
 
 impl MemoryStore {
     fn sync_connect(
-        namespace_memory_store: &NamespaceMemoryStore,
+        memory_stores: &mut MemoryStores,
         config: &MemoryStoreConfig,
         namespace: &str,
+        root_key: &[u8],
         kill_on_drop: bool,
     ) -> Result<Self, MemoryStoreError> {
         let max_stream_queries = config.common_config.max_stream_queries;
-        let namespace = namespace.to_string();
-        let store = namespace_memory_store
-            .get(&namespace)
-            .ok_or(MemoryStoreError::NotExistentNamespace)?;
+        ensure!(
+            memory_stores.namespaces.contains(namespace),
+            MemoryStoreError::NotExistentNamespace
+        );
+        let pair = (namespace.to_string(), root_key.to_vec());
+        let store = memory_stores.stores.entry(pair).or_insert_with(|| {
+            let map = MemoryStoreMap::new();
+            Arc::new(RwLock::new(map))
+        });
         let map = store.clone();
         let namespace = namespace.to_string();
+        let root_key = root_key.to_vec();
         Ok(MemoryStore {
             map,
             max_stream_queries,
             namespace,
+            root_key,
             kill_on_drop,
         })
     }
 
-    fn sync_list_all(namespace_memory_store: &NamespaceMemoryStore) -> Vec<String> {
-        namespace_memory_store.keys().cloned().collect::<Vec<_>>()
+    fn sync_list_all(memory_stores: &MemoryStores) -> Vec<String> {
+        memory_stores.namespaces.iter().cloned().collect::<Vec<_>>()
     }
 
-    fn sync_exists(namespace_memory_store: &NamespaceMemoryStore, namespace: &str) -> bool {
-        let namespace = namespace.to_string();
-        namespace_memory_store.contains_key(&namespace)
+    fn sync_exists(memory_stores: &MemoryStores, namespace: &str) -> bool {
+        memory_stores.namespaces.contains(namespace)
     }
 
-    fn sync_create(namespace_memory_store: &mut NamespaceMemoryStore, namespace: &str) {
-        let namespace = namespace.to_string();
-        let map = MemoryStoreMap::new();
-        let map = Arc::new(RwLock::new(map));
-        namespace_memory_store.insert(namespace, map);
+    fn sync_create(memory_stores: &mut MemoryStores, namespace: &str) {
+        memory_stores.namespaces.insert(namespace.to_string());
     }
```

Comment:
These could be methods of `MemoryStores` now.

Reply:
Yes, correct. That simplifies the code.